api/admin/controller.py

```diff
@@ -1107,13 +1107,27 @@ def _create_or_update_list(self, library, name, entries, collections, id=None):
         membership_change = False
         for entry in entries:
             pwid = entry.get("pwid")
-            work = self._db.query(
+            medium = entry.get("medium")
+            language = entry.get("language")
+
+            query = self._db.query(
                 Work
             ).join(
                 Edition, Edition.id==Work.presentation_edition_id
             ).filter(
                 Edition.permanent_work_id==pwid
-            ).one()
+            )
+
+            if medium:
+                query.filter(
+                    Edition.medium==Edition.additional_type_to_medium[medium]
+                )
+            if language:
+                query.filter(
+                    Edition.language==LanguageCodes.iso_639_2_for_locale(language)
+                )
```

Comment:
`query.filter` returns a new query, it doesn't modify the query, so you need to do `query = query.filter...`.

You could also use some tests for this code :)